miner/bid_simulator.go

```diff
@@ -593,6 +608,38 @@ func (b *bidSimulator) simBid(interruptCh chan int32, bidRuntime *BidRuntime) {
 		return
 	}
 
+	// fill transactions from mempool
+	if b.config.GreedyMergeTx {
+		delay := b.engine.Delay(b.chain, bidRuntime.env.header, &b.delayLeftOver)
+		if delay != nil && *delay > 0 {
+			log.Debug("BidSimulator: greedy merge tx stopTimer", "block", bidRuntime.env.header.Number,
+				"header time", time.Until(time.Unix(int64(bidRuntime.env.header.Time), 0)),
+				"commit delay", *delay, "DelayLeftOver", b.delayLeftOver)
+
+			stopTimer := time.NewTimer(*delay)
+
+			bidTxsSet := mapset.NewSet[common.Hash]()
+			for _, tx := range bidRuntime.bid.Txs {
```

Comment:
better to eliminate this for loop.
We can pass `bidRuntime.bid.Txs` into `fillTransactions(..., bidRuntime.bid.Txs)` directly.
```
// in fillTransactions(...)
for _, bigTx range bidRuntime.bid.Txs {
   // check and remove from pendingPlainTxs & pendingBlobTxs
}
```

Reply:
hashset makes it simple